packages/node_modules/overmind/src/utils.ts

```diff
@@ -29,9 +29,11 @@ export const makeStringifySafeMutations = (mutations: IMutation[]) => {
 
 export function deepCopy(obj) {
   if (isPlainObject(obj)) {
-    return Object.keys(obj).reduce((aggr, key) => {
-      aggr[key] = deepCopy(obj[key])
-
+    return Object.keys(obj).reduce((aggr: any, key) => {
+      if(obj.__lookupGetter__(key))
+        aggr.__defineGetter__(key, obj.__lookupGetter__(key))
+      else 
+        aggr[key] = deepCopy(obj[key])
```

Comment:
This is really great stuff! Though I would prefer if you use `Object.getOwnPropertyDescriptor` :-) Look at line 663 in index.ts, where it processes state. That way we are not using any "hacks" to redefine them.

But this is really excellent, great catch!

Reply:
Okay I updated it. 

I have noticed that we could use `if (isPlainObject(value) && !isAGetter) {` to simplify the logic in index that you referenced. However I am unsure if I should touch that code in this pr in another pr or we prefer not changing working code.